portal-impl/src/com/liferay/portal/service/impl/ReleaseLocalServiceImpl.java

```diff
@@ -188,47 +193,22 @@ public int getBuildNumberOrCreate() throws PortalException {
 			}
 		}
 
-		// Get release build number
-
-		try (Connection con = DataAccess.getConnection();
-			PreparedStatement ps = con.prepareStatement(
-				"select buildNumber from Release_ where releaseId = " +
-					ReleaseConstants.DEFAULT_ID);
-			ResultSet rs = ps.executeQuery()) {
-
-			int buildNumber = 0;
-
-			if (rs.next()) {
-				buildNumber = rs.getInt("buildNumber");
-			}
-			else {
-				buildNumber = _addReleaseInfo();
-			}
-
+		try {
+			db.runSQL("alter table Release_ add state_ INTEGER");
```

Comment:
@Preston-Crary the state_ adding was done in 2/28/2012, the schemaVersion adding was done in 8/22/2015. Both are very old, it should be safe to assume all customers having them already. Another way to look at it, we don't support that super old version directly upgrade to 7.2/7.3. If they do the upgrade by steps, they will have those columns added anyway. Let's just remove these lazy adding logic to keep things simple. Please resend, thanks!